### PR TITLE
added some solution between 3000-3010

### PR DESCRIPTION
**Pull request by agarwalhimanshugaya**
## Related Issue

Fix: #835 

## Description
I add some of the solution between 3000-3010 which is solved by mine 
## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa-solutions/lc-solutions/3000-3099/3000-Maximum area of diagonal rectangle.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+Return the area of the rectangle having the longest diagonal. If there are multiple rectangles with the longest diagonal, return the area of the rectangle having the maximum area.
+
+### Examples
+
+#### Example 1
+
+- **Input:** $dimensions = [[9,3],[8,6]]$
+- **Output:** $48$
+- **Explanation:** $For index = 0, length = 9 and width = 3. Diagonal length = sqrt(9 * 9 + 3 * 3) = sqrt(90) ≈ 9.487.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3000-3099/3000-Maximum area of diagonal rectangle.md:28]: replace

```md
$For index = 0, length = 9 and width = 3. Diagonal length = sqrt(9 * 9 + 3 * 3) = sqrt(90) ≈ 9.487.
For index = 1, length = 8 and width = 6. Diagonal length = sqrt(8 * 8 + 6 * 6) = sqrt(100) = 10.
So, the rectangle at index 1 has a greater diagonal length therefore we return area = 8 * 6 = 48..$
```

to 

```md
For index = $0$, $\text{length} = 9$ and width = $3$. $\text{Diagonal length} = sqrt(9 \times 9 + 3 \times 3) = sqrt(90) ≈ 9.487$.
For $\text{index} = 1$, $length = 8$ and $\text{width} = 6$. $\text{Diagonal length} = sqrt(8 \times 8 + 6 \times 6) = sqrt(100) = 10$.
So, the rectangle at index 1 has a greater diagonal length therefore we return $area = 8 \times 6 = 48..$
```

File: dsa-solutions/lc-solutions/3000-3099/3000-Maximum area of diagonal rectangle.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+#### Example 2
+
+- **Input:** $dimensions = [[3,4],[4,3]]$
+- **Output:** $12$
+- **Explanation:**$ Length of diagonal is the same for both which is 5, so maximum area = 12$.
+
+### Constraints
+
+- $1 <= dimensions.length <= 100$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3000-3099/3000-Maximum area of diagonal rectangle.md:40]: replace `$1 <= dimensions.length <= 100$` to `$1 \leq  dimensions.length \leq 100$`

File: dsa-solutions/lc-solutions/3000-3099/3000-Maximum area of diagonal rectangle.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        return max_area;
+
+    }
+};
+```
+
+### Conclusion
+
+At last we can say that we can calculate maximum area of diagonal rectangle using timecomplexity of o(n) and spacecomplexity of o(1) using a simple for loop and some variable
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3000-3099/3000-Maximum area of diagonal rectangle.md:118]: replace `o(n)` to `$O(n)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Update all files code base on one files suggestion and can't delete `dsa-solutions/lc-solutions/_category_.json` this file and you need write good format for all files based on suggestions

File: dsa-solutions/lc-solutions/3000-3099/3000-Maximum area of diagonal rectangle.md (hunk 1 above)
Comment (agarwalhimanshugaya) [reply dsa-solutions/lc-solutions/3000-3099/3000-Maximum area of diagonal rectangle.md:28]: how can i add again category_.json file

**Comment by agarwalhimanshugaya** [comment]
i apply all your given suggestions

**Comment by agarwalhimanshugaya** [comment]
i remove all the error mention by you

File: dsa-solutions/lc-solutions/3000-3099/3006-find beautiful indices in the given array.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+### Constraints
+
+- `1 <= k <= s.length <= 105`
+- `1 <= a.length, b.length <= 10`
+
+## Solution For the Problem Find Beautiful Indices in the Given Array I
+
+### Intuition
+
+Basically the problem statement suggest that we need to find the index of string a matching in string s and then check if there is any index j which is matching with b in s and abs(i - j) <= k.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3000-3099/3006-find beautiful indices in the given array.md:63]: reploace `abs(i - j) <= k` to 

```md
`abs(i - j) <= k`
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code only one error

**Review by Ajay-Dhangar — APPROVED** [review]
Done

File: dsa-solutions/lc-solutions/3000-3099/3005-count elements with maximum frequency.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
+---
+id: Count Elements With Maximum Frequency
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3000-3099/3005-count elements with maximum frequency.md:2]: replace `id: Count Elements With Maximum Frequency` to `id: count-elements-with-maximum-frequency`

File: dsa-solutions/lc-solutions/3000-3099/3010-divide an array into subarrays with minimum cost 1.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
+---
+id: Divide an Array Into Subarrays With Minimum Cost I
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3000-3099/3010-divide an array into subarrays with minimum cost 1.md:2]: replace `id: Divide an Array Into Subarrays With Minimum Cost I` to `id: divide-an-array-into-subarrays-with-minimum-cost-I`

File: dsa-solutions/lc-solutions/3000-3099/3006-find beautiful indices in the given array.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+So to find the matching index need calculate the longest prefix which is suffix array ( LPS array ) and in that if the lps[i] matches with pattern string length then we found a match - KMP algo.
+If you don't know KMP algo for pattern matching then first go through it then come back to solution.
+
+### Approach
+
+Get index of the matched pattern for the string a - lets call it v1
+Get index of the matched pattern for the string b - lets call it v2
+Check for possible combination of i & j in the v1 and v2.
+to do so we need to check if the abs(i - j). If so then simple add it to answer.
+If `v1[i] > v2[j]&& abs( i- j) > k` then keep moving pointer j till v1[i] > v2[j]
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3000-3099/3006-find beautiful indices in the given array.md:73]: replace `v1[i] > v2[j]` to 

```md
`v1[i] > v2[j]`
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Update code based on suggestions

**Comment by agarwalhimanshugaya** [comment]
mai push krrtha tha to nhi horha tha to muje merge krna pda origin k sath

**Comment by Ajay-Dhangar** [comment]
> mai push krrtha tha to nhi horha tha to muje merge krna pda origin k sath

okay

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Review by Ajay-Dhangar — APPROVED** [review]
...

**Pull request merged**